lib/Sema/CSGen.cpp

```diff
@@ -4082,10 +4061,8 @@ namespace {
 
       // Add explicit generic arguments, if there were any.
       if (expr->getGenericArgsRange().isValid()) {
-        if (addSpecializationConstraint(
-              CS.getConstraintLocator(expr), macroRefType,
-              expr->getGenericArgs()))
-          return Type();
+        addSpecializationConstraint(CS.getConstraintLocator(expr), macroRefType,
+                                    expr->getGenericArgs());
```

Comment:
Should use `addUnsolvedConstraint` here as well instead of completely ignoring the result.

Reply:
See above, the result was just for adding a note, but I've change this now.